src/R/Core/Impl/AST/Expressions/ExpressionParser.cs

```diff
@@ -383,6 +384,27 @@ enum OperationType {
         }
 
         private OperationType HandleOperator(ParseContext context, out ParseErrorType errorType) {
+            // We must distinguish between operator definition and the operator use.
+            // in case of %a% <- left side behaves as identifier rather than the operator
```

Comment:
`%a% <- ...` doesn't seem to be valid R. You have to write `` `%a%` <- ...`` for it to parse.

Similarly, ``1 `%a%` 2`` is actually not valid - you need to write ``1 %a% 2`` for it to parse as an operator.

So it should be sufficient to look at presence/absence of backticks to distinguish these two cases.